noir-projects/noir-protocol-circuits/crates/types/src/hash.nr

```diff
@@ -118,13 +120,33 @@ pub fn accumulate_sha256(input: [Field; 2]) -> Field {
     sha256_to_field(hash_input_flattened)
 }
 
-pub fn compute_logs_hash(previous_log_hash: Field, current_log_hash: Field) -> Field {
-    accumulate_sha256(
-        [
-        previous_log_hash,
-        current_log_hash
-    ]
-    )
+pub fn compute_tx_logs_hash<MAX_ENCRYPTED_LOGS_PER_TX>(logs: [SideEffect; MAX_ENCRYPTED_LOGS_PER_TX]) -> Field {
+    // TODO(Miranda): Below is flat hashing which would reduce constraints (we now only hash once in tail) - convert to this?
```

Comment:
Since we now only hash a large list of logs once (in the tail) - should we concat and hash rather than accumulate?
(This may cause issues with slices/array length however)

Reply:
To avoid a huge PR I will add this in the next one (requires fixtures/L1 contracts edits as well), alongside the tx.ts error, and adding a `logsCache` (like `noteCache`).